lib/ws_servers/api/send_authenticated.js

```diff
@@ -7,13 +7,15 @@ const decryptAPICredentials = require('../../util/decrypt_api_credentials')
 const openAuthBitfinexConnection = require('./open_auth_bitfinex_connection')
 const { DEFAULT_USER } = require('../../constants')
 
-module.exports = async (ws, db, d, opts) => {
+module.exports = async (ws, db, marketData, d, opts) => {
   const { authPassword, authControl } = ws
-  const { Market } = db
   const { mode } = opts
 
   const isPaper = mode === 'paper' ? 1 : 0
-  const markets = await Market.find([['p', '=', isPaper]])
+
+  const marketValues = [...marketData.values()]
+  const markets = marketValues.filter(m => m.p === isPaper)
```

Comment:
the same as in the comment above. In addition, for filtering `marketData`, you can take out a separate function in order to reduce the amount of boilerplate code (converting Map to an array)

Reply:
thanks!! added a general function to filter from the map object.